cmd/eval-dev-quality/cmd/evaluate_test.go

```diff
@@ -670,4 +672,143 @@ func TestEvaluateExecute(t *testing.T) {
 			filepath.Join("empty-response", "golang", "golang", "plain.log"): nil,
 		},
 	})
+
+	t.Run("Attempts", func(t *testing.T) {
+		{
+			queryMock := providertesting.NewMockQuery(t)
+			llmModel := llm.NewModel(queryMock, "testing-provider/testing-model")
+			providerMock := providertesting.NewMockProviderNamedWithModels(t, "testing-provider", []model.Model{llmModel})
+			validate(t, &testCase{
+				Name: "Single try fails",
+
+				Before: func(t *testing.T, logger *log.Logger, resultPath string) {
+					queryMock.On("Query", mock.Anything, "testing-provider/testing-model", mock.Anything).Return("", errors.New("empty response from model"))
+
+					provider.Register(
+						&struct {
+							provider.Provider
+							provider.Query
+						}{
+							Provider: providerMock,
+							Query:    queryMock,
+						},
+					)
+				},
+				After: func(t *testing.T, logger *log.Logger, resultPath string) {
+					delete(provider.Providers, "testing-provider")
```

Comment:
This is a source of problems, when we register a provider temporarily, we can run into concurrency problems. Better to refactor the code, make the provider a parameter and mock that parameter instead. Easier to type. Easier to understand. Less code. Cleaner.